pkgs/os-specific/linux/device-tree.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenvNoCC, dtc, findutils, raspberrypifw }:
```

Comment:
Somewhat related to @artemist's comment:

Nothing in `pkgs/os-specific/linux/` is directly at the root, everything ends up in sub-folder. The generic tooling implementation (`applyOverlays`) would be fine in `default.nix` in that sub-folder.

Additionally, the `kernel` folder can be used as an inspiration, where `linux-rpi.nix` provides the kernel for the raspberry pi family. The raspberry pi foundation's device tree things could be found under `pkgs/os-specific/linux/device-tree/raspberrypi.nix` for example.

Reply:
I've moved these files as well.